libraries/ESP8266WebServer/examples/FSBrowser/data/edit/index.htm

```diff
@@ -0,0 +1,1122 @@
+          }
+        }
+      }
+      
+      #header { 
+        position: absolute;
+        top: 0;
+        right: 0;
+        left: 0;
+        height:24px;
+        padding: 1px 1px 0px 10px;
+        background-color: #444;
+        color:#EEE;
+      }
+      #tree { 
+        position: absolute;
+        top: 25px;
+        bottom: 0;
+        left: 0;
+        width: 20%;
+        padding: 8px;
+      }
+      #editor, #preview { 
+        position: absolute;
+        top: 25px;
+        right: 0;
+        bottom: 0;
+        left: 20%;
+      }
+      #preview {
+        background-color: #EEE;
+        padding:5px;
+      }
+      
+      #status {
+        position: absolute;
+        top: 3px;
+        right: 10px;
+        font-size: 15px;
+      }
+      
+      #fsMeter {
+        width:100px;
+        padding-bottom:2px;
+      }
+      
+      #warning {
+        height:100%;
+        background-color:orange;
+        color:black;
+      }
+      .tooltip {
+        position: absolute;
+        z-index: 2; /* must be above editor which is at 1 */
+        right:0px;
+        top:20px;
+        visibility: hidden;
+        background-color: white;
+        color: black;
+        text-align: center;
+        border: 1px solid #000;
+        padding: 3px;
+        font-size: 10px;
+      }
+      #warning:hover .tooltip {
+        visibility: visible;
+      }
+
+      #loading {
+        position:absolute;
+        // note: changing between block and none cancels the opacity anim, so we move it in and out of screen instead by changing the top value
+        // maybe see https://www.impressivewebs.com/animate-display-block-none/ for another approach
+        display:block; 
+        top: -100vh;
+        left: 0;
+        width: 100vw;
+        height: 100vh;
+        z-index: 100;
+        background-color: rgba(0, 0, 0, 0.5);
+        opacity: 0;
+        transition: opacity 500ms ease-in-out;
+      }
+      #loading.shown {
+        top: 0;
+        opacity: 1;
+      }
+      
+      #loading-msg {
+        display: inline-block;
+        position: absolute;
+        top: 0px;
+        left: 50%;
+        transform: translateX(-50%);
+        color:white;
+        font-size: 32px;
+      }
+
+      @keyframes spinner-anim {
+        0% {
+          transform: rotate(0deg);
+        }
+        100% {
+          transform: rotate(360deg);
+        }
+      }
+      /* :not(:required) hides this rule from IE9 and below */
+      .spinner-anim:not(:required) {
+        display: inline-block;
+        position: relative;
+        top: 50%;
+        left: 50%;
+        transform: translate(-50%, -50%);
+        animation: spinner-anim 1s infinite linear;
+        border: 16px solid #eee;
+        border-right-color: transparent;
+        border-radius: 32px;
+        box-sizing: border-box;
+        overflow: hidden;
+        text-indent: -9999px;
+        width: 64px;
+        height: 64px;
+      }
+    </style>
```

Comment:
Does it make sense to move this out into a .css?

Reply:
Hmm. That's a good question.
For the sake of cleanliness, I'd move the css, js and the icons to separate files, but for ease of use, just dropping a single, self-contained, html file on the filesystem sounds much simpler.
It also makes compression easier, and makes the "embed html inside the program" relatively simple.
So all in all, I would leave it as is...